src/commands/compress.rs

```diff
@@ -44,20 +45,42 @@ pub fn compress_files(
                 // by default, ParCompress uses a default compression level of 3
                 // instead of the regular default that flate2 uses
                 gzp::par::compress::ParCompress::<gzp::deflate::Gzip>::builder()
-                    .compression_level(Default::default())
+                    .compression_level(
+                        level.map_or_else(Default::default, |l| gzp::Compression::new((l as u32).clamp(0, 9))),
+                    )
                     .from_writer(encoder),
             ),
-            Bzip => Box::new(bzip2::write::BzEncoder::new(encoder, Default::default())),
-            Lz4 => Box::new(lzzzz::lz4f::WriteCompressor::new(encoder, Default::default())?),
-            Lzma => Box::new(xz2::write::XzEncoder::new(encoder, 6)),
-            Snappy => Box::new(gzp::par::compress::ParCompress::<gzp::snap::Snap>::builder().from_writer(encoder)),
-            Zstd => {
-                let zstd_encoder = zstd::stream::write::Encoder::new(encoder, Default::default());
-                // Safety:
-                //     Encoder::new() can only fail if `level` is invalid, but Default::default()
-                //     is guaranteed to be valid
-                Box::new(zstd_encoder.unwrap().auto_finish())
-            }
+            Bzip => Box::new(bzip2::write::BzEncoder::new(
+                encoder,
+                level.map_or_else(Default::default, |l| bzip2::Compression::new((l as u32).clamp(1, 9))),
+            )),
+            Lz4 => Box::new(lzzzz::lz4f::WriteCompressor::new(
+                encoder,
+                lzzzz::lz4f::PreferencesBuilder::new()
+                    .compression_level(level.map_or(0, |l| (l as i32).clamp(1, lzzzz::lz4f::CLEVEL_MAX)))
+                    .build(),
+            )?),
+            Lzma => Box::new(xz2::write::XzEncoder::new(
+                encoder,
+                level.map_or(6, |l| (l as u32).clamp(0, 9)),
+            )),
+            Snappy => Box::new(
+                gzp::par::compress::ParCompress::<gzp::snap::Snap>::builder()
+                    .compression_level(gzp::par::compress::Compression::new(
+                        level.map_or_else(Default::default, |l| (l as u32).clamp(0, 9)),
+                    ))
+                    .from_writer(encoder),
+            ),
+            Zstd => Box::new(
+                zstd::stream::write::Encoder::new(
+                    encoder,
+                    level.map_or(0, |l| {
```

Comment:
here you can use `zstd::DEFAULT_COMPRESSION_LEVEL` directly, it's 3 instead of 0.
```suggestion
                    level.map_or(zstd::DEFAULT_COMPRESSION_LEVEL, |l| {
```